doc/contribute.rst

```diff
@@ -1,77 +1,129 @@
-------------------------
+If you start working on a new feature or a fix, please create an issue on
+GitHub shortly describing the issue and assign yourself.
+Your startpoint should always be the ``develop`` branch, which contains the
+lastest updates.
 
-To make pypesto easily usable, we are committed to documenting extensively.
-This involves in particular documenting the functionality of methods and
-classes, the purpose of single lines of code, and giving usage examples.
-The documentation is hosted on
-`pypesto.readthedocs.io <https://pypesto.readthedocs.io>`_
-and updated automatically every time the master branch on
-`github.com/icb-dcm/pypesto <https://github.com/icb-dcm/pypesto>`_ is updated.
-To compile the documentation locally, use::
+Create an own branch or fork, on which you can implement your changes. To
+get your work merged, please:
 
-    cd doc
-    make html
+1. create a pull request to the ``develop`` branch with a meaningful summary,
+2. check that code changes are covered by tests, and all tests pass,
+3. check that the documentation is up-to-date,
+4. request a code review from the main developers.
 
+Environment
+-----------
 
-Contribute tests
-----------------
+If you contribute to the development of pyPESTO, install developer requirements
+via::
 
-Tests are located in the ``test`` folder. All files starting with ``test_``
-contain tests and are automatically run on Travis CI. To run them manually,
-type::
+    pip install -r requirements-dev.txt
 
-    python3 -m pytest test
+Pre-commit hooks
+~~~~~~~~~~~~~~~~
 
-or alternatively::
+Firstly, this installs a `pre-commit <https://pre-commit.com/>`_ tool.
+To add those hooks to the `.git` folder of your local clone such that they are
+run on every commit, run::
 
-    python3 -m unittest test
+    pre-commit install
 
-You can also run specific tests.
+When adding new hooks, consider manually running ``pre-commit run --all-files``
+once as usually only the diff is checked. The configuration is specified in
+``.pre-commit-config.yaml``.
 
-Tests can be written with `pytest <https://docs.pytest.org/en/latest/>`_
-or the `unittest <https://docs.python.org/3/library/unittest.html>`_ module.
+Should it be necessary to perform commits without pre-commit verification,
+use ``git commit --no-verify`` or the shortform ``-n``.
 
+Tox
+~~~
 
-PEP8
-~~~~
+Secondly, this installs the virtual testing tool
+`tox <https://tox.readthedocs.io/en/latest/>`_, which we use for all tests,
+format and quality checks. Its configuration is specified in ``tox.ini``.
+To run it locally, simply execute::
 
-We try to respect the `PEP8 <https://www.python.org/dev/peps/pep-0008>`_
-coding standards. We run `flake8 <https://flake8.pycqa.org>`_ as part of the
-tests. If flake8 complains, the tests won't pass. You can run it via::
+    tox [-e flake8,doc]
+
+with optional ``-e`` options specifying the environments to run, see
+``tox.ini`` for details.
 
-    ./run_flake8.sh
+GitHub Actions
```

Comment:
Does this still belong to Environment?

Reply:
makes more sense stand-alone I think, agreed